driver/ppm_fillers.c

```diff
@@ -1513,9 +1515,26 @@ int f_proc_startupdate(struct event_filler_arguments *args)
 				mtime = file_inode(exe_file)->i_mtime.tv_sec * (uint64_t) 1000000000 + file_inode(exe_file)->i_mtime.tv_nsec;
 			}
 #endif
+			/* Before freeing the exefile we catch the resolved path for symlink resolution */
+			trusted_exepath = d_path(&exe_file->f_path, buf, PAGE_SIZE);
 			fput(exe_file);
 		}
 
+		/* The trusted_exepath could end with the suffix " (deleted)".
+		 * https://github.com/torvalds/linux/blob/2dde18cd1d8fac735875f2e4987f11817cc0bc2c/fs/d_path.c#L255
+		 * This is unhandy to manage in userspace, for this reason, we can remove it here
+		 */
+		if(trusted_exepath != NULL)
+		{
+			char deleted_suffix[] = " (deleted)";
+			int diff_len = strlen(trusted_exepath) - strlen(deleted_suffix);
+			if(diff_len > 0 &&
+				(strncmp(&trusted_exepath[diff_len], deleted_suffix, sizeof(deleted_suffix)) == 0))
+			{					
+				trusted_exepath[diff_len] = '\0';
+			}
+		}
```

Comment:
@FedeDP please take a look if here something is wrong we are dead :skull:

Reply:
:skull: